etna/models/seasonal_ma.py

```diff
@@ -183,12 +203,13 @@ def predict(self, ts: TSDataset, prediction_size: int, return_components: bool =
         ValueError:
             if forecast context contains NaNs
         """
-        if return_components:
-            raise NotImplementedError("This mode isn't currently implemented!")
-
         df = ts.to_pandas()
         new_df = self._predict(df=df, prediction_size=prediction_size)
         ts.df = new_df
+
+        if return_components:
+            target_components_df = self._predict_components(df=df, prediction_size=prediction_size)
```

Comment:
Why don't we have a logic with slice assignment here?

Reply:
I don't really understand. Does it mean that here we decompose true values instead of predicted values? Isn't it strange?

Reply:
No, we don't. We decompose predictions using the true values, because we use lags of true values to make predictions